packages/opencv-python/test_opencv_python.py

```diff
@@ -1,42 +1,42 @@
-import base64
+# flake8: noqa
+# mypy: disable-error-code=attr-defined
+
 import pathlib
+from ast import Compare
 
+import pytest
 from pytest_pyodide import run_in_pyodide
 
-REFERENCE_IMAGES_PATH = pathlib.Path(__file__).parent / "reference-images"
+REFERENCE_IMAGES_PATH = pathlib.Path(__file__).parent / "test_data"
 
 
-def compare_with_reference_image(selenium, reference_image, var="img", grayscale=True):
-    reference_image_encoded = base64.b64encode(reference_image.read_bytes())
-    grayscale = "cv.IMREAD_GRAYSCALE" if grayscale else "cv.IMREAD_COLOR"
-    match_ratio = selenium.run(
-        f"""
-        import base64
-        import numpy as np
+@pytest.fixture(autouse=True)
+def setup_compare_function(selenium):
+    @run_in_pyodide(packages=["opencv-python"])
+    def prepare(selenium):
         import cv2 as cv
-        DIFF_THRESHOLD = 2
-        arr = np.frombuffer(base64.b64decode({reference_image_encoded!r}), np.uint8)
-        ref_data = cv.imdecode(arr, {grayscale})
+        import numpy as np
+
+        def compare(ref, image, colormode):
+            DIFF_THRESHOLD = 2
+            MIN_MATCH_RATIO = 0.95
+
+            arr = np.frombuffer(ref, np.uint8)
+            ref_image = cv.imdecode(arr, colormode)
 
-        pixels_match = np.count_nonzero(np.abs({var}.astype(np.int16) - ref_data.astype(np.int16)) <= DIFF_THRESHOLD)
-        pixels_total = ref_data.size
-        float(pixels_match / pixels_total)
-        """
-    )
+            pixels_match = np.count_nonzero(
+                np.abs(image.astype(np.int16) - ref_image.astype(np.int16))
+                <= DIFF_THRESHOLD
+            )
+            pixels_total = image.size
 
-    # Due to some randomness in the result, we allow a small difference
-    return match_ratio > 0.95
+            match_ratio = float(pixels_match / pixels_total)
+            assert match_ratio > MIN_MATCH_RATIO
 
+        # TODO: this is a hack to make the function available in the global scope
+        np.compare_ = compare
```

Comment:
I wasn't able to find a better way to make a function or variable defined in some `run_in_pyodide` block visible in another `run_in_pyodide` block. Do you have any suggestions @hoodmane?

Reply:
Well I got a bit nerd sniped by this but I think I have a solution.

Reply:
See PR #49.